javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,401 @@
+"use strict";
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that iterates over an array of people objects
+and creates a new list of each person's full name using the array method 'map'.
+Each object will have the shape {firstName:string, lastName:string}
+E.g. [ { firstName:"Jane", lastName:"Doe" }, { firstName:"James", lastName:"Bond"}]
+should convert to ["Jane Doe", "James Bond"]
+Note the space in between first and last names.
+You can assume that neither firstName nor lastName will be blank
+------------------------------------------------------------------------------------------------ */
+const toLastNames = (people) => {
+  return people.map((obj) => obj.firstName + " " + obj.lastName);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that, given an array of numbers as input, uses reduce to add the values in the array.
+
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr) => {
+  let sum = arr.reduce((acc, curr) => {
+    acc += curr;
+    return acc;
+  }, 0);
+  return sum;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named addPurchases that, given an array of objects as input, uses reduce to find the total amount purchased. Each object contains the keys `item` and `purchasePrice` like the example.
+
+{
+  item: 'switch'
+  purchasePrice: 399
+},
+
+------------------------------------------------------------------------------------------------ */
+
+const addPurchases = (arr) => {
+  let total = arr.reduce((acc, curr) => {
+    acc += curr.purchasePrice;
+    return acc;
+  }, 0);
+  return total;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfElements that, given an array as input, uses reduce to count the number of elements in the array.
+
+Note: You may not use the array's built-in length property.
+------------------------------------------------------------------------------------------------ */
+
+const countNumberOfElements = (arr) => {
+  return arr.reduce((acc) => {
+    acc++;
+    return acc;
```

Comment:
Nice work.
You can simplify this even further, if you're so inclined.
```js
return arr.reduce(acc => acc++, 0);
```
Even if it works without it, I always like to put my starting value in. So having that "0" in there just keeps things clear. But sometimes the function will work as intended without it.